### PR TITLE
feat(builtins): introducing all builtins deployment

### LINKED ISSUE #42: Use frc42_dispatch once dependency issues are solved
## Description

The [frc42_dispatch](https://crates.io/crates/frc42_dispatch) crate should be used instead to replace hard-coded method numbers.

## Motivation

We sometimes hard coded some method numbers, computed in respect to FRC-0042 calling convention, making the code harder to extend and maintain.

A first example of such practice is in [`basic_test_actor`](https://github.com/polyphene/kythera/blob/f2017b318853acdb3e7080c521a100aa5fbbfc36/test_actors/actors/basic_test_actor/src/actor.rs#L29).

## Requirements

The [frc42_dispatch](https://crates.io/crates/frc42_dispatch) should be preferred. But at the time of writing, its use is unfortunately impossible as it creates dependency issues with other fvm-related crates.

## Open questions

N/A

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by tchataigner**
## Description

The main goal of this PR is to properly deploy all necessary builtins to our local execution environment, at the address they are expected to be based on the [builtin actors repository](https://github.com/filecoin-project/builtin-actors/blob/master/runtime/src/builtin/singletons.rs).

Aside from this, we also removed direct references to the builtins crates as per [this comment](https://github.com/filecoin-project/builtin-actors/issues/194#issuecomment-1316799088), only relying on it for the WASM bundle. Instead, to instantiate builtins and their state we use crates from [Chainsafe repository](https://github.com/ChainSafe/fil-actor-states).

Moreover, we now use `frc42_dispatch` crate in test actors. Closes #42 

Finally, we removed dependencies on the `ref-fvm` repository to instead be based on crates.io released version.  

## Notes

We should write some simple tests to check some flows that leverages builtin actors in the future (e.g.: Market actor, miner actor ...).

Added issue in backlog of the project: https://github.com/orgs/polyphene/projects/1/views/3?pane=issue&itemId=24909149

## Links to any relevant issues or pull requests

- #42

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/tests/lib.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        ;; Mock invoke function
+            (module
+                (func (export "invoke") (param $x i32) (result i32)
+                    (i32.const 1)
+                )
+            )
+        "#;
+
+#[test]
+fn test_builtin_deployed() {
```
Comment (PhilippeMts) [inline lib/tests/lib.rs:16]: great test. thanks for that.

should we work a bit on code duplication between this and tests in `lib/src/lib.rs` before moving forward?

File: test_actors/actors/builtins_test_actor/src/actor.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    INIT_ACTOR_ADDR, REWARD_ACTOR_ADDR, STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR,
+    SYSTEM_ACTOR_ADDR, VERIFIED_REGISTRY_ACTOR_ADDR,
+};
+use frc42_dispatch::match_method;
+use fvm_sdk as sdk;
+use fvm_sdk::actor::{get_actor_code_cid, get_builtin_actor_type};
+use fvm_sdk::NO_DATA_BLOCK_ID;
+use fvm_shared::error::ExitCode;
+
+// TODO use helix frc42_dispatch when their dependencies are up to date.
```
Comment (PhilippeMts) [inline test_actors/actors/builtins_test_actor/src/actor.rs:16]:
```suggestion
```

**Review by PhilippeMts — DISMISSED** [review]
Seems really good. Thanks for the encompassed issues and great progress here.
You'll find a couple of inline comments, but we are good to go imo. 🙏

File: lib/tests/lib.rs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        test_wasm_bin,
+        test_abi,
+    );
+
+    match tester.deploy_target_actor(target_actor) {
+        Err(_) => {
+            panic!("Could not set target Actor when testing if builtins are properly deployed")
+        }
+        _ => {}
+    }
```
Comment (jxs) [inline lib/tests/lib.rs:41]:
```suggestion
    tester.deploy_target_actor(target_actor).expect("Could not set target Actor when testing if builtins are properly deployed");
```

**Review by jxs — APPROVED** [review]
LGTM Thomas, only a couple small suggestions.

**Review by PhilippeMts — APPROVED** [review]
thanks for implementing this change 🙏 looks really great

**Pull request merged**